src/Symfony/Component/Translation/Dumper/QtFileDumper.php

```diff
@@ -33,6 +33,17 @@ public function formatCatalogue(MessageCatalogue $messages, $domain, array $opti
 
         foreach ($messages->all($domain) as $source => $target) {
             $message = $context->appendChild($dom->createElement('message'));
+            $metadata = $messages->getMetadata($source, $domain);
+            if (isset($metadata['sources'])) {
+                foreach ((array) $metadata['sources'] as $location) {
```

Comment:
`$metadata['sources']` can be a string or an array?

Reply:
Yes. See line 41: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/Tests/Dumper/PoFileDumperTest.php